pennylane/registers.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018-2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the :func:`registers` function, which builds registers for a given dictionary
+of registers. Registers are a dictionary of :class:`~.Wires` objects, where the key is the name of
+the register and the value is the ``Wires`` object.
+"""
+
+from .wires import Wires
+
+
+def registers(register_dict):
+    """Returns a collection of wire registers when given a dictionary of register names and sizes
+    (number of qubits in the register).
+
+    The registers are a dictionary of :class:`~.Wires` objects where the key is the register name
+    and the value is the ``Wires`` object. The individual wire labels are integers, ordered on
+    appearance, and nestedness.
+
+    Args:
+        register_dict (dict): a dictionary of registers where the keys are the names (str) of the
+            registers and the values are either nested dictionaries of more registers or
+            integers (int). At the most nested level for each register key, the value must be an
+            int.
+
+    Returns:
+        dict (Wires): dictionary where the keys are the names (str) of the registers, and the
```

Comment:
Better to be:
dict: dictionary where the keys are the names (str) of the registers and the values are either :class:`~.Wires` objects or other registers